datafusion/sql/src/unparser/ast.rs

```diff
@@ -360,6 +455,23 @@ impl RelationBuilder {
     pub fn has_relation(&self) -> bool {
         self.relation.is_some()
     }
+    pub fn get_name(&self) -> Option<&ast::ObjectName> {
+        match self.relation {
+            Some(TableFactorBuilder::Table(ref value)) => value.name.as_ref(),
+            _ => None,
+        }
+    }
+    pub fn get_alias(&self) -> Option<String> {
```

Comment:
You can probably avoid a bunch of copies if you made this return a reference to a `&str` rather than a `String` -- if the caller needed the string they can always copy it. 

```suggestion
    pub fn get_alias(&self) -> Option<&str> {
```

Reply:
It looks like `Ident` doesn't implement anything that would return a `&str`, so it needs a `String` intermediary. I'm also not sure what copies you're referring too, I don't make any copies of the values from `collect_valid_idents`? The return from `get_alias` also isn't cloned, and is taken ownership of by `collect_valid_idents`.